includes/payment-methods/class-wc-stripe-express-checkout-ajax-handler.php

```diff
@@ -92,9 +92,21 @@ public function ajax_add_to_cart() {
 		$product      = wc_get_product( $product_id );
 		$product_type = $product->get_type();
 
+		$booking_ids = [];
+		if ( 'booking' === $product_type ) {
+			$booking_ids = $this->express_checkout_helper->get_booking_ids_from_cart();
+		}
+
 		// First empty the cart to prevent wrong calculation.
 		WC()->cart->empty_cart();
 
+		// When a bookable product is added to the cart, a 'booking' is created with status 'in-cart'.
+		// This status is used to prevent the booking from being booked by another customer
+		// and should be removed when the cart is emptied for ECE purposes.
+		foreach ( $booking_ids as $booking_id ) {
+			do_action( 'wc-booking-remove-inactive-cart', $booking_id ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
+		}
```

Comment:
Is it worth running through the loop when that action has hooks attached?
```suggestion
		if ( has_action( 'wc-booking-remove-inactive-cart' ) ) { // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
			foreach ( $booking_ids as $booking_id ) {
				do_action( 'wc-booking-remove-inactive-cart', $booking_id ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
			}
		}
```